src/transformers/models/gpt_neo/configuration_gpt_neo.py

```diff
@@ -173,3 +178,97 @@ def num_attention_heads(self):
     @property
     def num_hidden_layers(self):
         return self.num_layers
+
+
+def custom_unfold(input, dimension, size, step):
+    """Custom torch.Tensor.unfold implementation to enable export to ONNX."""
+    import torch
+
+    shape = input.shape
+    rank = len(shape)
+    sizedim = shape[dimension]
+    low_indices = range(0, sizedim, step)
+    hi_indices = range(size, sizedim + 1, step)
+    stack = []
+    for low, hi in zip(low_indices, hi_indices):
+        s = [slice(None)] * rank
+        s[dimension] = slice(low, hi)
+        stack.append(input[s])
+    perm = list(range(0, rank))
+    perm.append(perm.pop(dimension))
+    unsqueeze = [t.permute(perm).unsqueeze(dimension) for t in stack]
+    return torch.cat(unsqueeze, dim=dimension)
```

Comment:
Are you sure this is being correctly traced? Generally, leaf variables resulting from Python operations are simply converted to a constant buffer. 

Imo, you should use `torch.arange()` , but this is supported. I would check within the graph this operations are not defined as constants. You can also try with different input shapes _(than the one used for exporting)_ potentially